src/packages/contrib/contrib.cc

```diff
@@ -3062,3 +3075,176 @@ void f_test_load() {
   }
 }
 #endif
+
+#ifdef F_HIGHEST
+
+void f_highest()
+{
+  int num = st_num_arg ;
+  int cnt = num - 1;
+
+  if( num == 0 )
+  {
+    error("highest() requires at least one argument of type int or float.\n");
+  }
+
+  // Sanity check to make sure that all arguments are the same type and that they
+  // are of valid types: int or float
+  do
+  {
+    if( (sp - cnt)->type != sp->type || ( (sp - cnt)->type != T_NUMBER && (sp - cnt)->type != T_REAL ) )
+    {
+      error("highest() All argument values must be of the same type (int or float).\n");
+    }
+  } while( cnt-- > 0 );
+
+  // If we only have one, just return it
+  if( num == 1 ) return ;
+
+  // Reset the counter
+  cnt = num - 1;
+
+  // If we're dealing with integers
+  if(sp->type == T_NUMBER)
+  {
+    LPC_INT highest = sp->u.number ;
+
+    do {
+      highest = ( sp - cnt )->u.number > highest ? ( sp - cnt )->u.number : highest ;
+    } while(cnt-- > 0);
+
+    pop_n_elems( num );
+    push_number( highest );
+  }
+  // If we're dealing with floats
+  else if(sp->type == T_REAL)
+  {
+    LPC_FLOAT highest = sp->u.real ;
+
+    do {
+      highest = ( sp - cnt )->u.real > highest ? ( sp - cnt )->u.real : highest ;
+    } while(cnt-- > 0);
+
+    pop_n_elems( num );
+    push_real( highest );
+  }
+  else
+  {
+    error("This is broken and should be reachable.") ;
+  }
+}
+
+#endif // F_HIGHEST
+
+#ifdef F_LOWEST
+void f_lowest()
+{
+  int num = st_num_arg ;
+  int cnt = num - 1;
+
+  if( num == 0 )
+  {
+    error("lowest() requires at least one argument of type int or float.\n");
+  }
+
+  // Sanity check to make sure that all arguments are the same type and that they
+  // are of valid types: int or float
+  do
```

Comment:
instead of several loops, probably make sense to just do one, also  I think it would be more usable to just compare everything as float and have it return with the same type of the value that is lowest.  for example

lowest(1, 0.9, 8) = 0.9
lowest (1, 2, 3) = 1
lowest (1 , 1.5, 1.9) = 1

would be great.

Reply:
C++ is not my strong suit, and I couldn't find an easy example where a u.number was converted to a u.real, and if I did that, do I push_number or push_real? I wasn't really confident in that. I agree that the comparison would be more usable how you describe, though.

Reply:
Actually, I think I found an appropriate example in `call_out.cc`

Reply:
Nevermind, I don't know what I'm doing. 🙄